sdk/storage/azure-storage-blob/tests/test_blob_client.py

```diff
@@ -559,5 +559,17 @@ def callback(response):
         custom_headers = {'User-Agent': 'customer_user_agent'}
         service.get_service_properties(raw_response_hook=callback, headers=custom_headers)
 
+    @GlobalStorageAccountPreparer()
+    def test_error_with_blank_conn_str(self, resource_group, location, storage_account, storage_account_key):
+        # Arrange
+
+        for service_type in SERVICES.items():
+            # Act
+            with self.assertRaises(ValueError) as e:
+                service = service_type[0].from_connection_string("", blob_name="test", container_name="container")
+            
+            self.assertEqual(
+                str(e.exception), "Connection string is either blank or malformed.")
```

Comment:
Could we also add something for the "malformed" scenario?
E.g., run the same loop for the following conn strs or any others you can think of:
```
for bad_conn_str in ["", "foo", "foo=bar=baz", "foo;bar;baz", "foo=;bar="]:
    for service_type in SERVICES.items():
        ...
```